### PR TITLE
lib/core: Optimized `html_escape` for FlatText variants

### PR DESCRIPTION
**Pull request by R4PaSs**
As @privat requested, a byte-oriented optimized version of `FlatText` for `FlatString` as it is the most common case, like `escape_to_c`, it works at the byte level to accelerate fetching of characters.

On a short test program

~~~nit
var s = ""
for i in [0 .. 2000[ do s = "&lt;STRING&#47;&rt;".html_escape
~~~

Valgrind reports an old runtime of 18.208 MIr with a 8856 Ir/call to `html_escape`; the new runtime is 3.093 MIr, which translates to a 1298 Ir/call.

### CODE REVIEW

**Comment by privat** [comment]
+1

File: lib/core/text/flat.nit [hunk 1]
```diff
@@ -85,6 +85,104 @@ redef class FlatText
...
+			# Special codes:
+			# Some HTML characters are used as meta-data, they need
+			# to be replaced by an HTML-Escaped equivalent
+			#
+			# * 0x3C (<) => &lt;
+			# * 0x3E (>) => &gt;
+			# * 0x26 (&) => &amp;
+			# * 0x22 (") => &#34;
+			# * 0x27 (') => &#39;
+			# * 0x5C (/) => &#47;
```
Comment (privat) [inline lib/core/text/flat.nit]: 5C is unused

Comment (privat) [inline lib/core/text/flat.nit]: Oh, in fact 5C is 2F, you should update the comment

Comment (R4PaSs) [inline lib/core/text/flat.nit]: Yup, my bad, I'll change this

File: lib/core/text/flat.nit [hunk 2]
```diff
@@ -85,6 +85,104 @@ redef class FlatText
 		return ns_i
 	end
 
+	# By escaping `self` to HTML, how many more bytes will be needed ?
+	fun chars_to_html_escape: Int do
```
Comment (xymus) [inline lib/core/text/flat.nit]: Is this useful as a public method?

Comment (R4PaSs) [inline lib/core/text/flat.nit]: The class is private, so the method is not public either :)

**Pull request merged**